packages/amplify_api_rest_plugin_interface/lib/src/RestOptions.dart

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/foundation.dart';
+import 'dart:typed_data';
+
+class RestOptions {
+  String apiName;
+  String path;
+  Uint8List body;
+  Map<String, String> queryParameters;
+  Map<String, String> headers;
+
+  RestOptions({
+    this.apiName,
+    @required this.path,
+    this.body,
+    this.queryParameters,
+    this.headers,
+  });
+
+  Map<String, dynamic> serializeAsMap() {
+    final Map<String, dynamic> restOptionsMap = <String, dynamic>{};
+
+    if (this.apiName != null) {
+      restOptionsMap["api_name"] = apiName;
+    }
+
+    if (this.path != null) {
+      restOptionsMap["path"] = path;
+    }
```

Comment:
`path` is required. We should probably throw an exception in the constructor if path is null

Reply:
We have the @required keyword in the constructor. 

What is the standard in our other plugins?  I thought they also just used @required.  I added this code to throw an exception: 

`    if(path.isEmpty) throw new Exception("Error no path provided to RestOptions"); 
`

Reply:
The `@required` keyword is only used by the linter (or dart analyzer) and not at compile time and run time. So if we are adding runtime validations, we should add when the `@required` param is passed and not when it's used.

Reply:
Okay sounds good.  Considering this conversation to be resolved then as we've added the exception if path is empty.